cns/restserver/internalapi.go

```diff
@@ -143,6 +146,68 @@ func (service *HTTPRestService) SyncNodeStatus(dncEP, infraVnet, nodeID string,
 	return
 }
 
+// SyncHostNCVersion will check NC version from NMAgent and save it as host NC version in container status.
+// If NMAgent NC version got updated, CNS will refresh the pending programming IP status.
+func (service *HTTPRestService) SyncHostNCVersion(ctx context.Context, channelMode string, lastUpdatedHostNCTimeStamp time.Time, forceMarkIPAvailableTimeRange time.Duration) {
+	var hostVersionNeedUpdateNcList []string
+	service.RLock()
+	for _, containerstatus := range service.state.ContainerStatus {
+		// Will open a separate PR to convert all the NC version related variable to int. Change from string to int is a pain.
+		hostNcVersion, err := strconv.Atoi(containerstatus.HostNCVersion)
+		if err != nil {
+			log.Errorf("Received err when change containerstatus.HostNCVersion %s to int, err msg %v", containerstatus.HostNCVersion, err)
+			return
+		}
+		dncNcVersion, err := strconv.Atoi(containerstatus.DncNCVersion)
```

Comment:
Can you compare it against containerStatus.Req.Version? I really dont know the usage of containerStatus.VMVersion and dont want to take dependency on it.

Reply:
OK